src/Shared/CoreCLRAssemblyLoader.cs

```diff
@@ -104,7 +104,9 @@ private Assembly TryResolveAssembly(AssemblyLoadContext context, AssemblyName as
                 {
                     foreach (var extension in _extensions)
                     {
-                        var candidatePath = Path.Combine(dependencyPath, $"{assemblyName.Name}.{extension}");
+                        var candidatePath = Path.Combine(dependencyPath,
+                            assemblyName.CultureName ?? string.Empty,
```

Comment:
Wikipedia [claims](https://en.wikipedia.org/wiki/Assembly_(CLI)) this is ok

> In general, library and process assemblies should be culture neutral; the culture should only be used for satellite assemblies.

But I have no idea how true that is. @nguerrera do you know?

I could make the subdirectory an _additional_ search path instead . . .